include/envoy/router/string_accessor.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+#include "envoy/request_info/filter_state.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Router {
+
+/**
+ * Contains a string in a form which is usable with FilterState and
+ * allows lazy evaluation if needed. All values meant to be accessible to the
+ * custom request/response header mechanism must use this type.
+ */
+class StringAccessor : public ::Envoy::RequestInfo::FilterState::Object {
```

Comment:
Our style is to leave off the `::Envoy:` in these cases.

Reply:
Ok, I'll fix that.  

Just so I can more usefully comment when I start doing envoy code reviews: Is there a section of the style guide that mandates this, or is this background custom?